lib/adapters/class-vip-enterprise-search.php

```diff
@@ -81,6 +81,12 @@ public function add_aggs_to_es_query( $formatted_args ): array {
 						$formatted_args['query']['function_score']['query']['bool']['filter'] ?? [],
 						$filter
 					);
+				} else {
+					// If we are using a search term, we need to add the filter to the post_filter.
```

Comment:
Not sure this is OK

Reply:
The logic above should have transformed queries with a search term to a function score query (see lines 52-56) and the elseif above this should put the filter in the proper place. Was that not working as expected? The risk with adding the `else` here is twofold: first, that it will only apply to results and not to the aggregations (e.g., that it will properly filter the search results, but the numbers in the aggregations won't be correct, like the number of items matching a particular post type or term); and second, because we're not checking the structure before augmenting it, there's a risk that we'll put this value in the wrong place.

Before making changes here, I'd like to understand why the existing logic doesn't work properly for applying the filter via the function score query.

Reply:
I re-reviewed this logic and it's not transforming in the way I was thinking, so I withdraw my objection here.